DataManagementSystem/DB/FileCatalogComponents/DirectoryTreeBase.py

```diff
@@ -336,148 +337,152 @@ def getDirectoryParameters( self, path ):
     """
-    result = self.db.ugManager.findUser( s_uid )
-    if not result['OK']:
-      return result
-    uid = result['Value']
-    result = self.db.ugManager.findGroup( s_gid )
-    if not result['OK']:
-      return result
-    gid = result['Value']
+    return self.changeDirectoryParameter( paths,
+                                          self.setDirectoryOwner,
+                                          self.db.fileManager.setFileOwner,
+                                          recursive = recursive )
 
-    result = checkArgumentFormat( paths )
-    if not result['OK']:
-      return result
-    arguments = result['Value']
-    successful = {}
-    failed = {}
-    for path, group in arguments.items():
-      result = self.setDirectoryGroup( path, group )
-      if not result['OK']:
-        failed[path] = result['Message']
-      else:
-        successful[path] = True
+#####################################################################
+  def changeDirectoryGroup( self, paths, recursive = False ):
+    """ Bulk setting of the directory group
 
-    return S_OK( {'Successful':successful, 'Failed':failed} )
+        :param dictionary paths : dictionary < lfn : group >
+    """
+    return self.changeDirectoryParameter( paths,
+                                          self.setDirectoryGroup,
+                                          self.db.fileManager.setFileGroup,
+                                          recursive = recursive )
 
 #####################################################################
   def setDirectoryMode( self, path, mode ):
-    """ set the directory mask
+    """ set the directory mode
+
+        :param mixed path: directory path as a string or int or list of ints or select statement
+        :param int mode: new mode
     """
     return self._setDirectoryParameter( path, 'Mode', mode )
 
 #####################################################################
-  def changeDirectoryMode( self, paths, s_uid = 0, s_gid = 0 ):
-    """ Bulk setting of the directory owner
+  def changeDirectoryMode( self, paths, recursive = False ):
+    """ Bulk setting of the directory mode
+
+        :param dictionary paths : dictionary < lfn : mode >
     """
-    result = self.db.ugManager.findUser( s_uid )
-    if not result['OK']:
-      return result
-    uid = result['Value']
-    result = self.db.ugManager.findGroup( s_gid )
-    if not result['OK']:
-      return result
-    gid = result['Value']
+    return self.changeDirectoryParameter( paths,
+                                          self.setDirectoryMode,
+                                          self.db.fileManager.setFileMode,
+                                          recursive = recursive )
 
-    result = checkArgumentFormat( paths )
-    if not result['OK']:
-      return result
-    arguments = result['Value']
+#####################################################################
+  def changeDirectoryParameter( self, paths,
```

Comment:
This method is absolutely not compatible with the changes I have done. This implements the recursive behavior at a higher level than the method actually changing the parameter (setDirectoryParameter), while I was propagating the recursive info till the very end for performance reasons